### PR TITLE
[core] Refactor GetFirstID() to be numeric-only

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Allows you to use:

```lua
        MEDUSA                = GetFirstID("Medusa") + 1,
        LIL_APKALLU           = GetFirstID("Lil_Apkallu") + 42,
```

if you so choose.

Also hits the db a lot less than before.

## Steps to test these changes

Works:

- First load, without errors on invalid zone names
- Hot-reload of IDs files

### CODE REVIEW

**Comment by TeoTwawki** [comment]
side note for naughty chilrens who look at this pr and then immediately set to using it:

Do not do this
```lua
        MEDUSA = GetFirstID("Medusa") ,
        ADD1   = GetFirstID("Medusa") + 1,
        ADD2   = GetFirstID("Medusa") + 2,
        ADD3   = GetFirstID("Medusa") + 3,
```

do this instead
```lua
        MEDUSA = GetFirstID("Medusa") ,
        ADD1   = MEDUSA + 1,
        ADD2   = MEDUSA + 2,
        ADD3   = MEDUSA + 3,
```

**Comment by zach2good** [comment]
Ready for review

File: src/map/lua/luautils.cpp [hunk 1]
```diff
@@ -847,81 +905,64 @@ namespace luautils
...
+        }
+        // clang-format on
+    }
+
+    void PopulateIDLookupsByZone(std::optional<uint16> maybeZoneId)
+    {
+        TracyZoneScoped;
+
+        // Make sure this has been run at least once!
+        auto result = lua.safe_script_file("scripts/globals/zone.lua");
```
Comment (claywar) [inline src/map/lua/luautils.cpp]: If zones are a straight enum, we could probably break that table out and move over there

**Review by claywar — APPROVED** [review]

File: src/map/lua/luautils.cpp (hunk 1 above)
Comment (zach2good) [reply src/map/lua/luautils.cpp]: Quite right, these aren't needed anymore - removed

**Pull request merged**